companion/src/modelslist.cpp

```diff
@@ -710,10 +720,46 @@ void TreeModel::refresh()
           availableEEpromSize -= size;
         }
       }
+      int protocol;
+      QString rxs;
+      for (unsigned j=0; j<CPN_MAX_MODULES; j++) {
+        protocol = model.moduleData[j].protocol;
+        // These are the only RXs that allow nominating RX # but changing RX or copying models can leave residual configuration which can cause issues
+        // if (protocol == PULSES_PXX_XJT_X16 || protocol == PULSES_PXX_XJT_LR12 || protocol == PULSES_PXX_R9M || protocol == PULSES_DSMX || protocol == PULSES_MULTIMODULE) {
+        if (!protocol == PULSES_OFF && model.moduleData[j].modelId > 0) {
+          if (!rxs.isEmpty()) {
+            rxs.append(", ");
+          }
+          unsigned mdlidx = model.moduleData[j].modelId;
+          rxs.append(QString("%1").arg(uint(mdlidx), 2, 10, QChar('0')));
+          if (!isModelIdUnique(mdlidx)) {
+            current->setHighlightRX(true);
+          }
+          ModelData & mdl = radioData->models[mdlidx-1];
+          if (mdl.isEmpty()) {
+            current->setHighlightRX(true);
+          }
+        }
+      }
+      current->setData(currentColumn++, rxs);
     }
   }
 
   if (hasEepromSizeData) {
     availableEEpromSize = (availableEEpromSize / 16) * 15;
   }
 }
+
+bool TreeModel::isModelIdUnique(unsigned modelIdx)
+{
+  int cnt = 0;
+  for (unsigned i=0; i<radioData->models.size(); i++) {
+    ModelData & model = radioData->models[i];
+    for (unsigned j=0; j<CPN_MAX_MODULES; j++) {
+      if (!model.moduleData[j].protocol == PULSES_OFF && model.moduleData[j].modelId == modelIdx) {
+        cnt++;
```

Comment:
you may return false from here to avoid looping on all models

Reply:
@bsongis yes will speed up. In testing the change I also found a bug and will fix both in the next commit.